sympy/series/limits.py

```diff
@@ -71,8 +70,16 @@ def heuristics(e, z, z0, dir):
         if r:
             rv = e.func(*r)
             if rv is S.NaN:
-                return
-
+                if e != ratsimp(e):
```

Comment:
There should be only one call of `ratsimp`. And it seems to me that it may raise an error under some conditions. So better wrap the single call in a try block to make the unit tests pass.